kinder/ci/tools/update-workflows/templates/workflows/rootless-tasks.yaml

```diff
@@ -0,0 +1,280 @@
+          echo "kube-scheduler is running as user kubeadm-ks"
+      else
+          echo "ERROR: kube-scheduler is not running as user kubeadm-ks"
+          res=1
+      fi
+
+      if pgrep kube-scheduler | xargs ps o group:16 --no-headers -p | grep -q kubeadm-ks ; then
+          echo "kube-scheduler is running as user kubeadm-ks"
+      else
+          echo "ERROR: kube-scheduler is not running as user kubeadm-ks"
+          res=1
+      fi
+
+      if pgrep etcd | xargs ps o user:16 --no-headers -p | grep -q kubeadm-etcd ; then
+          echo "etcd is running as user kubeadm-etcd"
+      else
+          echo "ERROR: etcd is not running as user kubeadm-etcd"
+          res=1
+      fi
+
+      if pgrep etcd | xargs ps o group:16 --no-headers -p | grep -q kubeadm-etcd ; then
+          echo "etcd is running as user kubeadm-etcd"
+      else
+          echo "ERROR: etcd is not running as user kubeadm-etcd"
+          res=1
+      fi
+
+      if [[ "\${res}" = 0 ]]; then
+        echo "All verify checks passed, congrats!"
+        echo ""
+      else
+        echo "One or more verify checks failed! See output above..."
+        echo ""
+        exit 1
+      fi
+      EOF
+
+      chmod +x /tmp/verify-rootless.sh
+- name: copy verify-rootless.sh on controlplane nodes
+  cmd: kinder
+  args:
+    - cp
+    - --name={{ .vars.clusterName }}
+    - /tmp/verify-rootless.sh
+    - "@cp*:/kinder/verify-rootless.sh"
+    - --loglevel=debug
+- name: init
+  description: |
+    Initializes the Kubernetes cluster with version "initVersion"
+    by starting the boostrap control-plane nodes
+  cmd: kinder
+  args:
+    - do
+    - kubeadm-init
+    - --name={{ .vars.clusterName }}
+    - --loglevel=debug
+    - --kubeadm-verbosity={{ .vars.kubeadmVerbosity }}
+    - --kubeadm-feature-gate="RootlessControlPlane=true"
+  timeout: 5m
+- name: join
+  description: |
+    Join the other nodes to the Kubernetes cluster
+  cmd: kinder
+  args:
+    - do
+    - kubeadm-join
+    - --name={{ .vars.clusterName }}
+    - --loglevel=debug
+    - --kubeadm-verbosity={{ .vars.kubeadmVerbosity }}
+  timeout: 10m
+- name: run verify-rootless.sh on controlplane nodes before upgrades
+  cmd: kinder
+  args:
+    - exec
+    - --name={{ .vars.clusterName }}
+    - "@cp*"
+    - /kinder/verify-rootless.sh
+    - --loglevel=debug
+- name: e2e-kubeadm
+  description: |
+    Runs kubeadm e2e tests
+  cmd: kinder
+  args:
+    - test
+    - e2e-kubeadm
+    - --test-flags=--report-dir={{ .env.ARTIFACTS }} --report-prefix=e2e-kubeadm
+    - --name={{ .vars.clusterName }}
+    - --loglevel=debug
+  timeout: 10m
+- name: e2e
+  description: |
+    Runs Kubernetes e2e test (conformance)
+  cmd: kinder
+  args:
+    - test
+    - e2e
+    - --test-flags=--report-dir={{ .env.ARTIFACTS }} --report-prefix=e2e
+    - --parallel
+    - --name={{ .vars.clusterName }}
+    - --loglevel=debug
+  timeout: 35m
```

Comment:
ok, so possibly i made i mistake in my steps summary but this should be moved after the rootless check after upgrade.
the rationale is " we care about the rootless tests before and after upgrade, but let's also run the e2e tests.

before the e2e tasks we should also add this "cluster info" task:
https://github.com/kubernetes/kubeadm/blob/master/kinder/ci/tools/update-workflows/templates/workflows/regular-tasks.yaml#L72-L80

- init / join
- rootless check
- upgrade
- rootless check
- cluster-info
- kubeadm / k8s e2e tests
- cleanup tasks

Reply:
Updated. The task order to match above.